Description: [Opus] Only include NEON sources and macros when arm_use_neon==true
Origin: upstream, https://chromium.googlesource.com/chromium/src/+/22d06b0c916aa422a57e58d634cb73c2909b320d

arm_neon.h throws an error when included in a non-neon build. Only
include neon-related sources when use_arm_neon is true.

---
 third_party/opus/BUILD.gn | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/third_party/opus/BUILD.gn b/third_party/opus/BUILD.gn
index 3bb5e0df499e..6211449f8a8a 100644
--- a/third_party/opus/BUILD.gn
+++ b/third_party/opus/BUILD.gn
@@ -371,12 +371,9 @@ static_library("opus") {
         "src/celt/arm/arm_celt_map.c",
         "src/celt/arm/armcpu.c",
         "src/celt/arm/armcpu.h",
-        "src/celt/arm/celt_neon_intr.c",
         "src/celt/arm/fft_arm.h",
         "src/celt/arm/mdct_arm.h",
         "src/celt/arm/pitch_arm.h",
-        "src/silk/arm/NSQ_neon.c",
-        "src/silk/arm/NSQ_neon.h",
         "src/silk/arm/arm_silk_map.c",
       ]
 
@@ -385,14 +382,25 @@ static_library("opus") {
       defines += [
         "OPUS_ARM_MAY_HAVE_EDSP",
         "OPUS_ARM_MAY_HAVE_MEDIA",
-        "OPUS_ARM_MAY_HAVE_NEON",
-        "OPUS_ARM_MAY_HAVE_NEON_INTR",
         "OPUS_HAVE_RTCD",
       ]
 
       deps = [
         ":convert_rtcd_assembler",
       ]
+
+      if (arm_use_neon) {
+        sources += [
+          "src/celt/arm/celt_neon_intr.c",
+          "src/silk/arm/NSQ_neon.c",
+          "src/silk/arm/NSQ_neon.h",
+        ]
+
+        defines += [
+          "OPUS_ARM_MAY_HAVE_NEON",
+          "OPUS_ARM_MAY_HAVE_NEON_INTR",
+        ]
+      }
     }
   }
 }
-- 
2.11.0

